spikeextractors/extractors/spykingcircusextractors/spykingcircusextractors.py

```diff
@@ -0,0 +1,154 @@
+from spikeextractors import SortingExtractor
+from spikeextractors.extractors.numpyextractors import NumpyRecordingExtractor
+import numpy as np
+from pathlib import Path
+
+try:
+    import h5py
+    HAVE_SCSX = True
+except ImportError:
+    HAVE_SCSX = False
+
+
+class SpykingCircusRecordingExtractor(NumpyRecordingExtractor):
+
+    extractor_name = 'SpykingCircusRecordingExtractor'
+    installed = True  # check at class level if installed or not
+    _gui_params = [
+        {'name': 'spykingcircus_folder', 'type': 'path', 'title': "Path to folder"},
+    ]
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, spykingcircus_folder):
+        spykingcircus_folder = Path(spykingcircus_folder)
+        listfiles = spykingcircus_folder.iterdir()
+        sample_rate = None
+        recording_file = None
+
+        parent_folder = None
+        result_folder = None
+        for f in listfiles:
+            if f.is_dir():
+                if any([f_.suffix == '.hdf5' for f_ in f.iterdir()]):
+                    parent_folder = spykingcircus_folder
+                    result_folder = f
+
+        if parent_folder is None:
+            parent_folder = spykingcircus_folder.parent
+            for f in parent_folder.iterdir():
+                if f.is_dir():
+                    if any([f_.suffix == '.hdf5' for f_ in f.iterdir()]):
+                        result_folder = spykingcircus_folder
+
+        assert isinstance(parent_folder, Path) and isinstance(result_folder, Path), "Not a valid spyking circus folder"
+
+        for f in parent_folder.iterdir():
+            if f.suffix == '.params':
+                sample_rate = _load_sample_rate(f)
+            if f.suffix == '.npy':
+                recording_file = str(f)
+        NumpyRecordingExtractor.__init__(self, recording_file, sample_rate)
```

Comment:
Are you sure that all .npy in SC are traces always ?

Reply:
No, but the our wrapper uses .npy. I'll leave as it is for now :)